### ISSUE TITLE
Virtual Offset to File Offset Gives Incorrect Address in Function parse_file_tab

### ISSUE THREAD

**Issue by alexander-hanel** [alexander-hanel/gopep#1]
When `ptr` is called in the below code it throws an exception because the virtual address to file offset is incorrectly converted.

```python 
    def parse_file_tab(self):
            try:
                file_tab_len = self.module_data.filetab_len
            except:
                return
            for c in range(4, file_tab_len + 1 ):
                offset = self.module_data.filetab + (c*4)
                # convert filetab to offset
                try:
                    file_tab_file_offset = self.pe.get_offset_from_rva(offset)
                except:
                    # calculate offset based off of already parsed
                    file_tab_file_offset = (self.module_data.filetab - self.module_data.ftab ) + self.gopclntab_offset + 12
                file_tab_offset = self.ptr(file_tab_file_offset + (c*4), size=4) + self.gopclntab_offset
                temp_string = self.pe_section[file_tab_offset:].split(b"\x00")[0]
                if temp_string:
                    self.filetab.append(temp_string)

```

- [x] Handle invalid pointer reads in `ptr`.  Quit execution or to continue on? 
- [x] Rewrite virtual offset to file offset. This current approach sucks. Keep in mind that some offset are relative to sections, modules data and the file start offset.

**Comment by alexander-hanel** [comment]
The conversion error is not very common. It has happened on a 31MB 64-bit executable but not on hello world 64 bit executable or similar 64 bit samples.

**Comment by alexander-hanel** [comment]
This calculation works for RVA to Offset. Super weird that I have to use `abs`. All of the example code I found was incorrect. 

```python 
import pefile, collections


class Convert():
    def __init__(self, data):
        self.DEBUG = True
        self.load_pe(data)

    def load_pe(self, pe_data):
        try:
            self.pe = pefile.PE(data=pe_data)

        except Exception as e:
            if self.DEBUG:
                print("ERROR: pefile load error %s") % (e)
            return

    def rva2file(self, offset):
        for section in self.pe.sections:
            if section.VirtualAddress < (offset - self.pe.OPTIONAL_HEADER.ImageBase) < (section.VirtualAddress + section.Misc_VirtualSize):
                return abs((self.pe.OPTIONAL_HEADER.ImageBase - offset) + ( section.VirtualAddress - section.PointerToRawData))
        return


with open("test.bin", "rb") as infile:
    pp = Convert(infile.read())

print(hex(pp.rva2file(0x01C470F0)))

```

**Comment by alexander-hanel** [comment]
Fixed with commit a75451e
